### DISCUSSION TITLE
Update dialog popup could be a notification

### DISCUSSION THREAD

**Discussion by xor-gate** [sparkle-project/Sparkle#2220] [Ideas]
Dear all,

[A user from the syncthing-macos project suggested](https://github.com/syncthing/syncthing-macos/issues/181) a nice idea to have Sparkle give a "push" notification in the notification center instead of a "nagging update dialog". As he finds it annoying when in the middle of work he gets a popup. I agree popups are really 2000 like, and macOS doesn't have much popups at all. I'm not sure which macOS version the notification center was introduced. The nice thing is when the notification area is used then user can manually accept the update the "old popup" is shown. And when the notifications are in Do-not-disturb mode they can be read be back later and the "workflow" is not interrupted with a software update popup dialog.

<img width="436" alt="Screenshot 2022-08-05 at 09 19 16" src="https://user-images.githubusercontent.com/1050166/183023936-02453cfa-1acb-4c96-ab3d-6bf64025a738.png">

The macOS notification API [is documented here](https://developer.apple.com/documentation/foundation/notifications)

**Comment by zorgiepoo** [comment]
Please check the release notes for Sparkle 2.2 which is aimed to resolve this and provides optional gentle scheduled updater APIs https://github.com/sparkle-project/Sparkle/releases/tag/2.2.0 and https://sparkle-project.org/documentation/gentle-reminders/

By default updates show up at better opportune times. Optionally (and very encouraged for background apps) developers can customize gentle reminders. The documentation and samples goes in depth including usage of user notifications.

**Reply by xor-gate** [reply]
Thanks for pointing this out, a quick look at it seems to be something I'm looking for. It is documented extensively.

**Reply by zorgiepoo** [reply]
> For backgrounded applications (apps that do not appear in the Dock), Sparkle 2.2 onwards will not let a scheduled update alert steal focus from another application or window that may be currently active – with the one exception of when your application just launched. Scheduled update alerts that show up after launch will be presented behind other apps and windows.

Note your application is a backgrounded one in the context of the documentation. Even if you don't do anything, Sparkle won't randomly steal focus from other apps anymore and may present alerts behind other apps/windows (except near launch / user initiation). Gentle reminders are for presenting a notification/badge/button-in-window/status icon/whatever in a less disruptive way than stealing focus.

**Reply by xor-gate** [reply]
Thanks for the info, indeed the application is a "tray icon" background application.
